File: Practica 4/Java/Figuras/FiguraGeometrica.java
package Figuras;

public class FiguraGeometrica {
    //Circulo
    double area(double radio){
        return Math.PI * radio * radio;
    }
    //Rectangulo
    double area(double base, double altura){
        return base * altura;
    }

    //Triangulo Rectangulo
    double area(double base, int altura){
        return (base * altura)/2;
    }
    //Trapecio
    double area(double baseMayor, double baseMenor, double altura){
        return ((baseMayor * baseMenor) * altura) / 2;
    }
    //Pentagono
    double area(int longitud, double apotema){
        return (5.0/2.0)*longitud * apotema;
    }

    public static void main(String[] args){
        FiguraGeometrica f1 = new FiguraGeometrica();
        FiguraGeometrica f2 = new FiguraGeometrica();
        FiguraGeometrica f3 = new FiguraGeometrica();
        FiguraGeometrica f4 = new FiguraGeometrica();
        FiguraGeometrica f5 = new FiguraGeometrica();


        System.out.println("Circulo: "+f1.area(1.0));
        System.out.println("Rectangulo: "+f2.area(2.0,3.0));
        System.out.println("Triangulo Rectangulo: "+f3.area(3.0,7));
        System.out.println("Trapecio: "+f4.area(2.0,3.0,5.0));
        System.out.println("Pentagono: "+f5.area(3,5.0));
    }
}

File: Practica 4/Python/FiguraGeometrica.py
#pip install multimethod
from multimethod import multimethod
import math
class FiguraGeometrica:
    #Circulo
    @multimethod
    def area(self,radio:float):
        return math.pi*radio**2
    #Triangulo Rectangulo
    @multimethod
    def area(self, base:float, altura:float):
        return base * altura

    #Rectangulo
    @multimethod
    def area(self, base: float, altura: int):
        return (base * altura) / 2

    #Trapecio
    @multimethod
    def area(self, baseMayor:float, baseMenor:float, altura:float):
        return ((baseMayor + baseMenor) * altura) / 2

    #Pentagono
    @multimethod
    def area(self, longitud:int, apotema:float):
        return (5 / 2) * longitud * apotema

Figuras = FiguraGeometrica()
f1 = Figuras.area(1.0)
f2 = Figuras.area(2.0,3.0)
f3 = Figuras.area(3.0,7)
f4 = Figuras.area(2.0,3.0,5.0)
f5 = Figuras.area(3,5.0)

print("Circulo: ",f1)
print("Rectangulo: ", f2)
print("Triangulo rectangulo: ", f3)
print("Trapecio: ", f4)
print("Pentagono: ", f5)

